### PR TITLE
Wishlist functionality

### PR DESCRIPTION
**Pull request by sudipkundu999**
In this PR :
- Fixing product context and reducer logic
- Adding add to wishlist functionality
  - **works even if a user is not logged in**

[Preview Link](https://deploy-preview-8--cobra-store-react.netlify.app/products)
A bug exists: _The toast is shown twice on adding/removing from the wishlist_

### CODE REVIEW

**Review by parulgupta2001 — APPROVED** [review]
Remove commented code otherwise everything  looks good to me.

**Review by parulgupta2001 — APPROVED** [review]

File: src/components/ProductFilter.jsx [hunk 1]
```diff
@@ -86,7 +90,7 @@ export const ProductFilter = (props) => {
               name="category"
               value={item}
               onChange={(e) => {
-                setSearchParams("");
+                // setSearchParams("");
```
Comment (iRohitGaur) [inline src/components/ProductFilter.jsx:94]: remove commented code

**Review by iRohitGaur — APPROVED** [review]
UI looks great. Nice work! Good to merge 👍🏻

File: src/components/ProductFilter.jsx (hunk 1 above)
Comment (sudipkundu999) [reply src/components/ProductFilter.jsx:94]: This code is for future use.

**Comment by sudipkundu999** [comment]
The bug still exists. I will fix that in subsequent PRs.

**Pull request merged**